### PR TITLE
[ADD] 14.0 pos_reuse_pricelist module

### PR DESCRIPTION
**Pull request by PierrickBrun**

### CODE REVIEW

File: pos_reuse_pricelist/static/src/js/models.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    models.PosModel = models.PosModel.extend({
+        on_removed_order(removed_order, index, reason) {
+            posmodel_super.on_removed_order.apply(this, arguments);
+            var order_list = this.get_order_list();
+            if (
+                (reason === "abandon" || removed_order.temporary) &&
+                order_list.length > 0
+            ) {
+                // Do nothing when this is a deletion (and there is another order available)
```
Comment (ivantodorovich) [inline pos_reuse_pricelist/static/src/js/models.js]: Better to inverse that `if` statement and remove this empty block

File: pos_reuse_pricelist/static/src/js/models.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/*
+ *  Copyright 2021 Akretion
+ *  License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+ */
+
+odoo.define("pos_reuse_pricelist.models", function (require) {
+    "use strict";
+
+    var models = require("point_of_sale.models");
```
Comment (ivantodorovich) [inline pos_reuse_pricelist/static/src/js/models.js]:
```suggestion
    const models = require("point_of_sale.models");
```

Replace all `var` with `const`

https://www.freecodecamp.org/news/var-let-and-const-whats-the-difference/

File: pos_reuse_pricelist/static/src/js/models.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+                order_list.length > 0
+            ) {
+                // Do nothing when this is a deletion (and there is another order available)
+            } else {
+                // Reuse the pricelist
+                this.get_order().pricelist = removed_order.pricelist;
+            }
+        },
+    });
+});
```
Comment (ivantodorovich) [inline pos_reuse_pricelist/static/src/js/models.js:27]:
```suggestion
    return models;
});
```

This is to play nice with other modules, making it easier to extend this one

File: pos_reuse_pricelist/static/src/js/models.js (hunk 1 above)
Comment (PierrickBrun) [reply pos_reuse_pricelist/static/src/js/models.js]: The idea was to reuse the same `if` statement as the one in `posmodel_super.on_removed_order` to make it easier to understand and maintain.
I think the inversed statement may be harder to read

Comment (ivantodorovich) [reply pos_reuse_pricelist/static/src/js/models.js]: Something like this maybe?
Could be accompanied with a comment

```js
    const newOrderCreated = !((reason === "abandon" || removed_order.temporary) && order_list.length > 0);
    if (newOrderCreated) {
        this.get_order().pricelist = removed_order.pricelist;
    }
```

Anyway, not blocking :)

**Review by ivantodorovich — APPROVED** [review]

File: pos_reuse_pricelist/static/src/js/models.js (hunk 1 above)
Comment (PierrickBrun) [reply pos_reuse_pricelist/static/src/js/models.js]: Good idea

**Review by legalsylvain — APPROVED** [review]
Lgtm

**Pull request merged**